src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -929,6 +941,14 @@ public function __call($method, $parameters)
 		{
 			return $this->callScope($scope, $parameters);
 		}
+		elseif ( ! is_null($this->model['fillable']) && in_array($column = 'is_'.$method, $this->model['fillable']))
```

Comment:
Thanks @GrahamCampbell, I definitely see the efficiency improvements - but I'm unsure about the readability as I don't think it is instantly obvious why `fillable` being something other than `null` should then lead to a check for the other conditions inside the `elseif`. Whereas in it's current form the ` ! is_null` reads as just another check and offers less importance, to me.

But that may just be due to my understanding of why fillable would be `null`. I only added the `null` check as an after-thought so that it passed the tests, as with my current knowledge of the framework I couldn't see any real-life cases were fillable would be `null` unless explicitly set.

I'm happy to go with the consensus and update the code if you wish, completely agree readability is important.